File: Baekjoon/Problem_23000~23999/Problem_23756/Problem_23756.cpp
#include <cstdio>
#include <algorithm>

using namespace std;

int main() {
    int N; scanf("%d", &N);
    int A[100]; for (int i = 0; i <= N; i++) scanf("%d", &A[i]);

    int _sum = 0;
    for (int i = 0; i < N; i++) {
        _sum += min(abs(A[i] - A[i+1]), min(abs(A[i] + 360 - A[i+1]), abs(A[i] - A[i+1] - 360)));
    }
    
    printf("%d\n", _sum);
    
    return 0;
}

File: Baekjoon/Problem_23000~23999/Problem_23756/Problem_23756.py
import sys

input = lambda: sys.stdin.readline().rstrip()

if __name__ == "__main__":
    N = int(input())
    A = [ int(input()) for _ in range(N+1) ]
    
    _sum = 0
    for i in range(N):
        _sum += min([abs(A[i] - A[i+1]), abs(A[i] - (A[i+1] + 360)), abs(A[i] + 360 - A[i+1])])
    print(_sum)